WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/DaysProvider.kt

```diff
@@ -0,0 +1,33 @@
+package org.wordpress.android.ui.bloggingreminders
+
+import org.wordpress.android.fluxc.model.BloggingRemindersModel.Day
+import org.wordpress.android.fluxc.model.BloggingRemindersModel.Day.FRIDAY
+import org.wordpress.android.fluxc.model.BloggingRemindersModel.Day.MONDAY
+import org.wordpress.android.fluxc.model.BloggingRemindersModel.Day.SATURDAY
+import org.wordpress.android.fluxc.model.BloggingRemindersModel.Day.SUNDAY
+import org.wordpress.android.fluxc.model.BloggingRemindersModel.Day.THURSDAY
+import org.wordpress.android.fluxc.model.BloggingRemindersModel.Day.TUESDAY
+import org.wordpress.android.fluxc.model.BloggingRemindersModel.Day.WEDNESDAY
+import org.wordpress.android.ui.reader.utils.DateProvider
+import java.util.Calendar
+import javax.inject.Inject
+
+class DaysProvider
+@Inject constructor(private val dateProvider: DateProvider) {
+    private val days = listOf(
+            Calendar.SUNDAY to SUNDAY,
+            Calendar.MONDAY to MONDAY,
+            Calendar.TUESDAY to TUESDAY,
+            Calendar.WEDNESDAY to WEDNESDAY,
+            Calendar.THURSDAY to THURSDAY,
+            Calendar.FRIDAY to FRIDAY,
+            Calendar.SATURDAY to SATURDAY
```

Comment:
Even though this works pretty well (as the logic inside `DateProvider` does too), we should probably avoid using `Calendar` around this feature once #14698 gets merged and use `DayOfWeek` instead, since that's already being used in some places.

I know those parts of the code don't interact with each other right now, but I'm just trying to be on the careful side, since those two APIs have some significant differences when it comes to days of week (`Calendar.SUNDAY == 1` and `DayOfWeek.SUNDAY == 7`).


If we decide to start using desugaring within FluxC as well, then I guess this would actually be a requirement.

For now, I'd just add a `// TODO` on the top of this block to remind us of doing that. Let me know what you think!

Reply:
that's a good point. I agree 👍 . We can probably add a TODO and create a task to rewrite this once that PR is merged 👍